test/unit/strings_spec.lua

```diff
@@ -8,6 +8,42 @@ local to_cstr = helpers.to_cstr
 local strings = cimport('stdlib.h', './src/nvim/strings.h',
                         './src/nvim/memory.h')
 
+describe('vim_strsave_escaped()', function()
+  local vim_strsave_escaped = function(s, chars)
+    local res = strings.vim_strsave_escaped(to_cstr(s), to_cstr(chars))
+    local ret = ffi.string(res)
+
+    -- Explicitly free memory so we are sure it is allocated: if it was not it 
+    -- will crash.
+    strings.xfree(res)
+    return ret
+  end
+
+  it('precedes by a backslash all chars from second argument', function()
+    eq('\\a\\b\\c\\d', vim_strsave_escaped('abcd','abcd'))
```

Comment:
you can avoid the extra escapes by using lua's `[[ ]]` syntax to indicate a literal string.

    eq([[\a\b\c\d]], vim_strsave_escaped('abcd','abcd'))